torchio/transforms/preprocessing/spatial/resample.py

```diff
@@ -183,64 +161,122 @@ def apply_transform(self, subject: Subject) -> Subject:
                     matrix = matrix.numpy()
                 image.affine = matrix @ image.affine
 
-            floating_itk = image.as_sitk(force_3d=True)
+            floating_sitk = image.as_sitk(force_3d=True)
+
+            resampler = sitk.ResampleImageFilter()
+            resampler.SetInterpolator(interpolator)
+            self._set_resampler_reference(
+                resampler,
+                self.target,
+                floating_sitk,
+                subject,
+            )
+            resampled = resampler.Execute(floating_sitk)
+
+            array, affine = sitk_to_nib(resampled)
+            image.set_data(torch.as_tensor(array))
+            image.affine = affine
+        return subject
 
-            # Get reference image
-            if isinstance(self.reference_image, str):
+    def _set_resampler_reference(
+            self,
+            resampler: sitk.ResampleImageFilter,
+            target: Union[TypeSpacing, TypePath, Image],
+            floating_sitk,
+            subject,
+            ):
+        # Target can be:
+        # 1) An instance of torchio.Image
+        # 2) An instance of pathlib.Path
+        # 3) A string, which could be a path or an image in subject
+        # 3) A string, which could be a path or an image in subject
+        # 4) A number or sequence of numbers for spacing
+        # 5) A tuple of shape, affine
+        # The fourth case is the different one
+        if isinstance(target, (str, Path, Image)):
+            if Path(target).is_file():
+                # It's an existing file
+                path = target
+                image = ScalarImage(path)
+            elif isinstance(target, Image):
+                # It's a TorchIO image
+                image = target
```

Comment:
it was used before by RandomAnisotropy, but I wonder if still usefull ?

Reply:
Hmm I think it's still good to leave it, especially for backward compatibility.

Reply:
sure no problem, I was just wondering if you see other usecase